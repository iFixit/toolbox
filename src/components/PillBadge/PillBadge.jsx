import React from 'react';
import { string } from 'prop-types';
import glamorous from 'glamorous';
import constants from '../../constants';

const { spacing, fontSize, lineHeight, color } = constants;

const PillBadgeContainer = glamorous.span(props => ({
   display: 'inline-block',
   padding: `${spacing[0]} ${spacing[1]}`,
   fontSize: fontSize[0],
   fontWeight: 700,
   lineHeight: lineHeight.none,
   textTransform: 'uppercase',
   color: props.color,
   backgroundColor: props.background,
   borderRadius: 9999,
}));

const PillBadge = props => <PillBadgeContainer {...props} />;

PillBadge.propTypes = {
   /** Set the background color. */
   background: string,
   /** Set the text color. */
   color: string,
};

PillBadge.defaultProps = {
   background: color.grayAlpha[1],
   color: color.grayAlpha[6],
};

export default PillBadge;
