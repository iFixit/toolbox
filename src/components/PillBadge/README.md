### Examples

#### Default PillBadge

```
<PillBadge>42</PillBadge>
```

#### Alternate colors

Remember to use the colors defined in the `constants` object instead of hard-coding HEX or RGBA values.

```
// NOTE: These colors are hard-coded for demo purposes only. Don't hard-code HEX values in production code.

<PillBadge color="#C27E00" background="#FEF5D9">12K</PillBadge>
```
