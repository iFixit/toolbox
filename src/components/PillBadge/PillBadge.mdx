---
name: PillBadge
route: /components/pill-badge
menu: Components
---

import PillBadge from './PillBadge'
import { Playground, PropsTable } from 'docz'

# PillBadge

## Examples

### Default PillBadge

<Playground>
  <PillBadge>42</PillBadge>
</Playground>

#### Alternate colors

Remember to use the colors defined in the `constants` object instead of hard-coding HEX or RGBA values.

<Playground>
  {/* NOTE: These colors are hard-coded for demo purposes only. Don't hard-code HEX values in production code.a */}
  <PillBadge color="#C27E00" background="#FEF5D9">12K</PillBadge>
</Playground>

## Props

<PropsTable of={PillBadge} />
