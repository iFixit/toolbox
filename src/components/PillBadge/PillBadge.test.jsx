import React from 'react';
import renderer from 'react-test-renderer';
import PillBadge from './PillBadge';

test('renders without crashing', () => {
   const tree = renderer.create(<PillBadge>Test</PillBadge>).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with custom text color and background color', () => {
   const tree = renderer
      .create(
         <PillBadge color="#C27E00" background="#FEF5D9">
            Test
         </PillBadge>,
      )
      .toJSON();
   expect(tree).toMatchSnapshot();
});
