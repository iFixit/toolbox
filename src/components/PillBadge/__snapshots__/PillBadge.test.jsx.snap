// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom text color and background color 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 0 0.5rem;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1.5;
  text-transform: uppercase;
  color: #C27E00;
  background-color: #FEF5D9;
  border-radius: 9999px;
}

<span
  className="glamor-0"
>
  Test
</span>
`;

exports[`renders without crashing 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 0 0.5rem;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1.5;
  text-transform: uppercase;
  color: rgba(0, 3, 6, 0.54);
  background-color: rgba(0, 3, 6, 0.07);
  border-radius: 9999px;
}

<span
  className="glamor-0"
>
  Test
</span>
`;
